crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -187,3 +188,121 @@ async fn volume_protocol_fee_sell_order() {
 
     protocol_fee_test_case(test_case).await;
 }
+
+#[tokio::test]
+#[ignore]
+async fn price_improvement_fee_buy_out_of_market_order() {
+    let fee_policy = FeePolicy::PriceImprovement {
+        factor: 0.5,
+        max_volume_factor: 1.0,
+        quote: PriceImprovementQuote {
+            sell_amount: 50000000000000000000u128.into(),
+            buy_amount: 35000000000000000000u128.into(),
+            network_fee: 1000000000000000000u128.into(),
+        },
+    };
+    let order_sell_amount = 50000000000000000000u128.into();
+    let order_buy_amount = 40000000000000000000u128.into();
+    let test_case = TestCase {
+        order_side: order::Side::Buy,
+        fee_policy,
+        order_sell_amount,
+        network_fee: Some(2000000000000000000u128.into()),
```

Comment:
This will likely be easier to see with #2472, but can you comment in code how the price improvement is achieved (realized network fee lower than quote, leading to X price improvement). This will help understand the expected executed sell amount (compared to the quoted one), but maybe you can add a comment for this expectation as well.

Should be done for each test.

Reply:
Started writing the comments and found out that the results look pretty weird. Found [this code](https://github.com/cowprotocol/services/blob/8f8ef93bf6bdf27a77995f1897559010dbd8d004/crates/driver/src/tests/setup/blockchain.rs#L97-L110) which applies [a default surplus](https://github.com/cowprotocol/services/blob/8f8ef93bf6bdf27a77995f1897559010dbd8d004/crates/driver/src/tests/cases/mod.rs#L24) factor. Not sure if it needs to be eliminated for the price improvement tests.

Reply:
Yes, either we need to set it in the test case or make sure it's set to 1. I think your trick with reducing the fee is good enough and allows us to more precisely reason about absolute surplus values (rather than a factor), so I'd slightly prefer setting the factor to 1 (but no strong feelings, as longs as it's clear from the main test how the calculation is carried out)